### PR TITLE
Fixes 223

### LINKED ISSUE #223: Improvments for Admin code
- Add `ORDER_BY` clause in list commands
- Use job factory in tests to make them more stable (btw, make unspecified parameters in job factory random)
- Add queueing lock info in the shell

(ping @entwanne FYI)

### PR DESCRIPTION
**Pull request by ewjoachim**
Closes #223 

### Successful PR Checklist:

- [x] Tests
  - [ ] (not applicable?)
- [ ] Documentation
  - [x] (not applicable?)
- [x] Had a good time contributing?

### CODE REVIEW

File: procrastinate/admin.py [hunk 1]
```diff
@@ -47,39 +48,44 @@ async def list_jobs_async(
...
+            )
+        ]
 
     async def list_queues_async(
-        self, queue: str = None, task: str = None, status: str = None, lock: str = None,
+        self,
+        queue: Optional[str] = None,
+        task: Optional[str] = None,
+        status: Optional[str] = None,
+        lock: Optional[str] = None,
```
Comment (entwanne) [inline procrastinate/admin.py:88]: Doesn't this one need a `queueing_lock` parameter as well?

File: procrastinate/shell.py [hunk 2]
```diff
@@ -96,9 +97,9 @@ def do_retry(self, arg):
 
         Example: retry 2
         """
-        print_job(self.admin.set_job_status(arg, status="todo"))
+        print_job(self.admin.set_job_status(arg, status="todo"))  # type: ignore
```
Comment (entwanne) [inline procrastinate/shell.py:100]: Why do you need to ignore typing here?

File: procrastinate/admin.py [hunk 3]
```diff
@@ -101,29 +107,30 @@ async def list_queues_async(
...
+            )
+        ]
 
     async def list_tasks_async(
-        self, queue: str = None, task: str = None, status: str = None, lock: str = None,
+        self,
+        queue: Optional[str] = None,
+        task: Optional[str] = None,
+        status: Optional[str] = None,
+        lock: Optional[str] = None,
```
Comment (entwanne) [inline procrastinate/admin.py:133]: Same here?

File: procrastinate/admin.py (hunk 1 above)
Comment (ewjoachim) [reply procrastinate/admin.py:88]: I'm not sure it makes sense... What information would you be searching if you listed queues with a queueing_lock filter ?

Happy to add it if you have a use case or if you think the lack of consistency in this aspect would be confusing.

File: procrastinate/shell.py (hunk 2 above)
Comment (ewjoachim) [reply procrastinate/shell.py:100]: mypy undestands `set_job_status_async` but not `set_job_status`.

https://github.com/peopledoc/procrastinate/issues/101

File: procrastinate/admin.py (hunk 3 above)
Comment (ewjoachim) [reply procrastinate/admin.py:133]: Same answer :)

File: procrastinate/admin.py (hunk 1 above)
Comment (entwanne) [reply procrastinate/admin.py:88]: No it was just for the consistency with `list_jobs`.

**Pull request merged**